### PR TITLE
[WaitForARMFeedback] Long Term Retention for Managed Instances 

### PR DESCRIPTION
**Pull request by xaliciayang**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json [hunk 1]
```diff
@@ -16,64 +16,6 @@
     "application/json"
   ],
   "paths": {
-    "/subscriptions/{subscriptionId}/providers/Microsoft.Sql/locations/{locationName}/managedDatabaseRestoreAzureAsyncOperation/{operationId}/completeRestore": {
```
Comment (pixia) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json]: why is this included?

Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json]: This is removed

**Review by jaredmoo — APPROVED** [review]

**Review by pixia — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedDatabases.json [hunk 2]
```diff
@@ -638,6 +639,13 @@
           "x-ms-mutability": [
             "create"
           ]
+        },
```
Comment (allenjzhang) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedDatabases.json]: Is there a reason that 2019-06-01-preview version does not have this and LongTermRetentionManagedInstanceBackups.json?

Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedDatabases.json]: Added managedDatabases.json to 2019-06-01-preview as well.  the Backups/Policies APIs are new and we are going to use 2018-06-01-preview for those.

**Comment by allenjzhang** [comment]
Code style issues has been found in PR. Please follow [this link](https://aka.ms/AA6h31t) to fix the issues. Thanks!

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json [hunk 3]
```diff
@@ -0,0 +1,920 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Sql/locations/{locationName}/longTermRetentionManagedInstances/{managedInstanceName}/longTermRetentionDatabases/{databaseName}/longTermRetentionManagedInstanceBackups/{backupName}": {
```
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json]: can these backups exist at both scopes - RG and subscription? can they be created at both these scopes? Why is the support needed for subscription scope? To use these, the caller will need /write /read permissions at the scope of a subscription which is more privileged than getting the permissions at a RG scope. 
A regular database is created under an RG.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json [hunk 4]
```diff
@@ -0,0 +1,920 @@
...
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
```
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json]: WHat about the PUT api for this? how are these created?

Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json]: These are created through the currently existing backup infrastructure.  Generally, full database backups are taken 1x/week and our workflow determines whether or not a full backup for a server/managed instance should be saved for long term retention.

**Comment by ravbhatnagar** [comment]
both the questions were addressed. Signing off from ARM side.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json (hunk 3 above)
Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json]: Backups can exist with or without a RG. 

In cases of LTR backups tied to dropped instances, the resource will not have an RG: subscriptions/$subscriptionId/providers/Microsoft.Sql/Locations/$location/longTermRetentionManagedInstances/$instanceName/longTermRetentionDatabases/$databaseName/longTermRetentionManagedInstanceBackups/$backupName. 

However, backups belonging to live instances _will_ contain a RG.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json (hunk 4 above)
Comment (xaliciayang) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json]: In other words, a user cannot "create" a new backup
